### PR TITLE
Update dependency versions and update node versions of CI targets.

### PR DESCRIPTION
**Pull request by sttk**
This pr is based on #94, and updates versions of dependencies and test targets for reducing vulnerability prints by npm as possible and preparing to modify to fix a gulp-cli's bug.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (phated) [inline .eslintrc]: let's use the `extends: "gulp"` eslintrc for now.  It requires eslint@^2 though.

File: package.json [hunk 2]
```diff
@@ -16,26 +16,27 @@
     "LICENSE"
   ],
   "scripts": {
-    "test": "jshint lib index.js && jscs lib index.js && mocha -t 5000 -b -R spec test/index",
-    "cover": "nyc --reporter=lcov --reporter=text-summary npm test"
+    "lint": "eslint .",
+    "test": "jshint lib index.js && mocha -t 5000 -b -R spec test/index",
```
Comment (phated) [inline package.json]: remove jshint, create a `pretest` script that calls the linter (like https://github.com/gulpjs/sparkles/blob/master/package.json#L21)

File: package.json [hunk 3]
```diff
@@ -16,26 +16,27 @@
...
+    "cover": "eslint . && nyc --reporter=lcov --reporter=text-summary npm test"
```
Comment (phated) [inline package.json]: if linting is done as a `pretest` it won't need to exist as part of this command.

File: package.json [hunk 4]
```diff
@@ -16,26 +16,27 @@
...
   },
   "dependencies": {
     "extend": "^3.0.0",
-    "findup-sync": "^2.0.0",
+    "findup-sync": "^3.0.0",
```
Comment (phated) [inline package.json]: major bump probably requires a major bump from us?  I forget why I major bumped that module.

File: package.json [hunk 5]
```diff
@@ -16,26 +16,27 @@
...
     "fined": "^1.0.1",
     "flagged-respawn": "^1.0.0",
     "is-plain-object": "^2.0.4",
     "object.map": "^1.0.0",
-    "rechoir": "^0.6.2",
+    "rechoir": "^0.7.0",
```
Comment (phated) [inline package.json]: I think something breaking happened in that bump.

File: package.json [hunk 6]
```diff
@@ -16,26 +16,27 @@
...
     "resolve": "^1.1.7"
   },
   "devDependencies": {
     "chai": "^3.5.0",
     "coffeescript": "^1.10.0",
-    "jscs": "^3.0.7",
+    "eslint": "^5.14.1",
```
Comment (phated) [inline package.json]: use version `^2`

File: test/lib/register_loader.js [hunk 7]
```diff
@@ -37,19 +37,47 @@ describe('registerLoader', function() {
       registerLoader(app, extensions, configPath);
     });
 
+    it('Should emit only a "require" event when registering loader ' +
```
Comment (phated) [inline test/lib/register_loader.js]: if you add an `.eslintrc` in the test directory that extends `gulp/test`, you don't need to worry about long test names.

File: test/lib/register_loader.js [hunk 8]
```diff
@@ -37,19 +37,47 @@ describe('registerLoader', function() {
...
+          expect(moduleName).to.be.equal('xxx');
+          expect(error).to.be.an('error');
+          expect(error.message).to.contain('Cannot find module');
+        } else if (index === 1) {
+          expect(moduleName).to.be.equal(loaderPath);
+          expect(error).to.be.an('error');
+          expect(error.message).to.contain('Cannot find module');
+          done();
+        } else {
+          expect.fail();
```
Comment (phated) [inline test/lib/register_loader.js]: instead of `expect.fail()`, I'd prefer `done(new Error("some message"))`.

**Comment by phated** [comment]
@sttk just a few comments.  I think we can inline this with the patterns I was doing with the other gulpjs projects (like https://github.com/gulpjs/sparkles)

File: .eslintrc (hunk 1 above)
Comment (sttk) [reply .eslintrc]: First I used it but eslint output errors because js-liftoff uses `const`.

Comment (phated) [reply .eslintrc]: @sttk yeah, I had to update the `const` to `var` in other modules as well.  If you don't have the time to do that, I can try to do it this weekend.

File: package.json (hunk 4 above)
Comment (sttk) [reply package.json]: It seems there is no breaking changes. Updating is-glob and normalize-path.
I think this bump is good.

File: package.json (hunk 5 above)
Comment (sttk) [reply package.json]: Breaking is by [rechoir#37](https://github.com/gulpjs/rechoir/pull/37) ? Is it better to defer this bump until liftoff's major bump?

Comment (phated) [reply package.json]: Yeah, I think we should defer that.  I'm still nervous we didn't catch all the edge cases with that feature.

**Comment by sttk** [comment]
@phated I've modified as you pointed out except using 'eslint-config-gulp'.
But it's ok to replace this pr by your update.

**Comment by phated** [comment]
This was rebased and merged.

Closed by 7853fd33d6f97c2abb5f81b7197736f0e0a1d971

I also converted the project to `eslint-config-gulp` at c0fc65fdfc2ca3cc3dcc6fa241b938496be5df7c

Thanks @sttk!

**Comment by sttk** [comment]
@phated Thanks!

**Pull request closed without merging**